### PR TITLE
feat: spelled-number pattern & since duration linter

### PR DESCRIPTION
**Pull request by hippietrail**
# Issues 

#404 

# Description

Implements #404 for non-native English speakers, especially German speakers who write "I tried since two days" instead of "I tried for two days" or "I tried since two days ago".

This required a new pattern for numbers written in words. For now this only goes from `one` to `ninety-nine`.
Compounds are recognized with or without hyphens so `ninety nine` also works.

I wasted a little bit of time on failing tests where I was using `exact_word`. Not for the first time I wasn't sure whether `exact` means case-sensitive. It does so I added a doc comment making that clear. And I added tests for `exact_word` and `any_capitalization_of` to ensure the case sensitivity of each.

# Demo

# How Has This Been Tested?

I added unit tests for the new pattern and for the new linter.
The linter patterns test edge cases and several sentences sourced from GitHub.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests to cover my changes

### CODE REVIEW

File: harper-core/src/linting/since_duration.rs [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+    fn correct_ago_all_caps() {
+        assert_top3_suggestion_result(
+            "BUG: SINCE 6 MONTHS UNLOAD CHECKPOINT",
+            SinceDuration::default(),
+            "BUG: SINCE 6 MONTHS AGO UNLOAD CHECKPOINT",
+        );
+    }
+
+    #[test]
+    #[ignore = "We can't yet handle modifiers like 'over'. Plus it doesn't work with 'ago'."]
```
Comment (elijah-potter) [inline harper-core/src/linting/since_duration.rs:271]: Do you intend to come back to this in the future and improve support?

File: harper-core/src/linting/since_duration.rs [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+            "Same! For two days.",
+        );
+    }
+
+    #[test]
+    fn correct_what_changed() {
+        assert_top3_suggestion_result(
+            "What changed since two weeks?",
+            SinceDuration::default(),
+            "What changed since two weeks ago?",
```
Comment (elijah-potter) [inline harper-core/src/linting/since_duration.rs:194]: Nice!

File: harper-core/src/linting/since_duration.rs [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+
+#[cfg(test)]
+mod tests {
+    use crate::linting::SinceDuration;
+    use crate::linting::tests::{assert_lint_count, assert_top3_suggestion_result};
+
+    #[test]
+    fn catches_spelled() {
+        assert_lint_count(
+            "I have been waiting since two hours.",
```
Comment (elijah-potter) [inline harper-core/src/linting/since_duration.rs:111]: I adore this test coverage.

**Review by elijah-potter — APPROVED** [review]
